internal/extract/json.go

```diff
@@ -122,32 +126,35 @@ func Into(body interface{}, to interface{}) error {
 	return err
 }
 
-// IntoStructPtr will unmarshal the given body into the provided
-// interface{} (to).
-func IntoStructPtr(body, to interface{}, label string) error {
+func typeCheck(to interface{}, kind reflect.Kind) error {
 	t := reflect.TypeOf(to)
 	if k := t.Kind(); k != reflect.Ptr {
 		return fmt.Errorf("expected pointer, got %v", k)
 	}
-	switch t.Elem().Kind() {
-	case reflect.Struct:
-		return intoPtr(body, to, label)
-	default:
-		return fmt.Errorf("expected pointer to struct, got: %v", t)
+
+	if kind != t.Elem().Kind() {
+		return fmt.Errorf("expected pointer to %v, got: %v", kind, t)
```

Comment:
`kind` is an int, it will be logged as an int (`25` for `Struct`)

Reply:
done